Dashboard/extract_combined.py

```diff
@@ -0,0 +1,50 @@
+"""Combined Extract Script: Identifies the store name and executes the relevant extraction"""
+import importlib
+import re
+
+
+STORE_LIST = ['asos']
+
+
+def identify_store(product_url: str) -> str | None:
+    """Returns the store from the given URL."""
+    store_patterns = {
+        r'^(https?://)?(www\.)?asos\.[a-zA-Z]{2,}/.*': 'Asos'}
+
+    for pattern, store_name in store_patterns.items():
+        if re.match(pattern, product_url):
+            return store_name
+
+    print("Store not identified")
+    return None
+
+
+def extract_product_information(product_url: str) -> tuple:
+    """Given an URL identifies the store and calls the relevant extract file."""
+    store_name = identify_store(product_url)
+
+    if store_name:
+        if store_name.lower() in STORE_LIST:
+            filename = f"extract_from_{store_name.lower()}"
+            try:
+
+                spec = importlib.util.spec_from_file_location(
+                    filename, f"{filename}.py")
+                module = importlib.util.module_from_spec(spec)
+                spec.loader.exec_module(module)
+                print(module.extract_product_information(product_url))
+            except FileNotFoundError:
+                print(f"Extraction module '{filename}.py' not found.")
+            except AttributeError:
+                print(
+                    f"Module '{filename}' does not have 'extract_product_information' function.")
```

Comment:
I think this can be simplified a lot. I think it would be easier to instead import the `extract_product_information` function from `extract_from_asos.py` and call it directly. For instance, you could have a dictionary that stores all of these functions:
```
extract_functions = {
  'asos' : extract_from_asos.extract_product_information
}
```
Here, `extract_functions` contains the functions to extract from a particular website.

You could also do this as a function:
these functions:
```
def extract_from(store_name: str, product_url: str):
  if store_name == 'asos':
    return extract_from_asos.extract_product_information(product_url)
...
```

I think either choice is fine but the second one is a bit more to the point